Support streaming in Azure ML and few naming changes

sachinparyani opened the pull request:
- [x] **PR title**: "community: Support streaming in Azure ML and few naming changes"

- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** Added support for streaming for azureml_endpoint. Also, renamed `LlamaContentFormatter` to `OpenAIStyleContentFormatter`, `LlamaChatContentFormatter` to `OpenAIStyleChatContentFormatter` and `AzureMLEndpointApiType.realtime` to `AzureMLEndpointApiType.dedicated`


- [ ] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.


- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, hwchase17.

Commit:
Support streaming in Azure ML andnaming changes

Commit:
Support streaming in Azure ML andnaming changes

Commit:
fix issue with async streaming

Commit:
fix issue with async streaming

sachinparyani commented on libs/community/langchain_community/llms/azureml_endpoint.py:
```diff
@@ -440,7 +442,7 @@ class AzureMLOnlineEndpoint(BaseLLM, AzureMLBaseEndpoint):
         .. code-block:: python
             azure_llm = AzureMLOnlineEndpoint(
                 endpoint_url="https://<your-endpoint>.<your_region>.inference.ml.azure.com/score",
```
for dedicated api types, I think we still expect the endpoint url to be /score and not /chat/completions

sachinparyani replied:
@santiagxf can you confirm?

santiagxf replied:
Not 100% of the models support the OpenAI-compatible protocol, so we should keep having `score` which is supportted across all of them.

santiagxf commented on docs/docs/integrations/chat/azureml_chat_endpoint.ipynb:
```diff
@@ -137,7 +137,7 @@
     "    endpoint_api_type=AzureMLEndpointApiType.serverless,\n",
     "    endpoint_api_key=\"my-api-key\",\n",
-    "    content_formatter=LlamaChatContentFormatter,\n",
+    "    content_formatter=OpenAIStyleChatContentFormatter,\n",
```
Is "style" the right word? Can we see if there is a better naming concensus in the ecosystem? For instance, `litellm` uses the name `CustomOpenAI`.

santiagxf commented on docs/docs/integrations/llms/azure_ml.ipynb:
```diff
@@ -181,7 +181,7 @@
+    "    endpoint_api_type=\"dedicated\",\n",
     "    endpoint_api_key=os.getenv(\"BART_ENDPOINT_API_KEY\"),\n",
     "    endpoint_url=os.getenv(\"BART_ENDPOINT_URL\"),\n",
     "    model_kwargs={\"temperature\": 0.8, \"max_new_tokens\": 400},\n",
```
Wondering if we should make this strongly typed so we help the developer, but cast to dictionary in case the user wants a custom parameter. Something like `OpenAIInferenceParams(BaseInferenceParams)` and `BaseInferenceParams(dict)`.

sachinparyani replied:
For now lets keep the same enum. As per our earlier discussion, the goal was to eventually get rid of it anyway because we intend to close the gap on the client side between dedicated and serverless.

leo-gan committed:
docs: API Reference builder bug fix (#17890)

Issue in the API Reference:
If the `Classes` of `Functions` section is empty, it still shown in API
Reference. Here is an
[example](https://api.python.langchain.com/en/latest/core_api_reference.html#module-langchain_core.agents)
where `Functions` table is empty but still presented.
It happens only if this section has only the "private" members (with
names started with '_'). Those members are not shown but the whole
member section (empty) is shown.

liugddx committed:
docs: update sparkllm intro doc (#17848)

**Description:** update sparkllm intro doc.
**Issue:** None
**Dependencies:** None
**Twitter handle:** None

volodymyr-memsql committed:
community[patch]: Added add_images method to SingleStoreDB vector store (#17871)

In this pull request, we introduce the add_images method to the
SingleStoreDB vector store class, expanding its capabilities to handle
multi-modal embeddings seamlessly. This method facilitates the
incorporation of image data into the vector store by associating each
image's URI with corresponding document content, metadata, and either
pre-generated embeddings or embeddings computed using the embed_image
method of the provided embedding object.

the change includes integration tests, validating the behavior of the
add_images. Additionally, we provide a notebook showcasing the usage of
this new method.

---------

Co-authored-by: Volodymyr Tkachuk <vtkachuk-ua@singlestore.com>

qqubb committed:
docs: minor grammatical correction. (#17724)

- **Description:** a minor grammatical correction.

gianlucagiudice committed:
Docs: Fix typo (#17733)

- **Description:** fix doc typo

ccurme committed:
core: fix .bind when used with RunnableLambda async methods (#17739)

**Description:** Here is a minimal example to illustrate behavior:
```python
from langchain_core.runnables import RunnableLambda

def my_function(*args, **kwargs):
    return 3 + kwargs.get("n", 0)

runnable = RunnableLambda(my_function).bind(n=1)


assert 4 == runnable.invoke({})
assert [4] == list(runnable.stream({}))

assert 4 == await runnable.ainvoke({})
assert [4] == [item async for item in runnable.astream({})]
```
Here, `runnable.invoke({})` and `runnable.stream({})` work fine, but
`runnable.ainvoke({})` raises
```
TypeError: RunnableLambda._ainvoke.<locals>.func() got an unexpected keyword argument 'n'
```
and similarly for `runnable.astream({})`:
```
TypeError: RunnableLambda._atransform.<locals>.func() got an unexpected keyword argument 'n'
```
Here we assume that this behavior is undesired and attempt to fix it.

**Issue:** https://github.com/langchain-ai/langchain/issues/17241,
https://github.com/langchain-ai/langchain/discussions/16446

mackong committed:
community[patch]: add missing chunk parameter for _stream/_astream (#17807)

- Description: Add missing chunk parameter for _stream/_astream for some
chat models, make all chat models in a consistent behaviour.
- Issue: N/A
- Dependencies: N/A

ztoliver committed:
community[patch]: Allow override of 'fetch_schema_from_transport' in the GraphQL tool (#17649)

- **Description:** In order to override the bool value of
"fetch_schema_from_transport" in the GraphQLAPIWrapper, a
"fetch_schema_from_transport" value needed to be added to the
"_EXTRA_OPTIONAL_TOOLS" dictionary in load_tools in the "graphql" key.
The parameter "fetch_schema_from_transport" must also be passed in to
the GraphQLAPIWrapper to allow reading of the value when creating the
client. Passing as an optional parameter is probably best to avoid
breaking changes. This change is necessary to support GraphQL instances
that do not support fetching schema, such as TigerGraph. More info here:
[TigerGraph GraphQL Schema
Docs](https://docs.tigergraph.com/graphql/current/schema)
  - **Threads handle:** @zacharytoliver

---------

Co-authored-by: Zachary Toliver <zt10191991@hotmail.com>
Co-authored-by: Bagatur <baskaryan@gmail.com>

nvoxland-al committed:
docs: Improved deeplake.py init documentation (#17549)

**Description:** 
Updated documentation for DeepLake init method.

Especially the exec_option docs needed improvement, but did a general
cleanup while I was looking at it.

**Issue:** n/a
**Dependencies:** None

---------

Co-authored-by: Nathan Voxland <nathan@voxland.net>

Speuce committed:
docs: Fix URL typo in tigris.ipynb (#17894)

- **Description:** The URL in the tigris tutorial was htttps instead of
https, leading to a bad link.
  - **Issue:** N/A
  - **Dependencies:** N/A
  - **Twitter handle:** Speucey

baskaryan commented on libs/community/langchain_community/llms/azureml_endpoint.py:
```diff
@@ -53,12 +53,12 @@ def call(
     """
 
-    realtime = "realtime"
+    dedicated = "dedicated"
```
this is a breaking change. can we keep realtime and dedicated and treat them the same (and update all the docs to use dedicated)?

Commit:
community[patch]: add possibility to search by vector in OpenSearchVectorSearch (#17878)

- **Description:** implements the missing `similarity_search_by_vector`
function for `OpenSearchVectorSearch`
- **Issue:** N/A
- **Dependencies:** N/A

rgupta2508 committed:
community[patch]: added SCANN index in default search params (#17889)

This will enable users to add data in same collection for index type
SCANN for milvus

cbornet committed:
docs: Update langchain-astradb README with AstraDBStore (#17864)

sepiatone committed:
docs: Change module import path for SQLDatabase in the documentation (#17874)

**Description:** This PR changes the module import path for SQLDatabase
in the documentation
**Issue:** Updates the documentation to reflect the move of integrations
to langchain-community

leo-gan committed:
docs: modules descriptions (#17844)

Several `core` modules do not have descriptions, like the
[agent](https://api.python.langchain.com/en/latest/core_api_reference.html#module-langchain_core.agents)
module.
- Added missed module descriptions. The descriptions are mostly copied
from the `langchain` or `community` package modules.

cbornet committed:
docs: AstraDB VectorStore docstring (#17834)

leo-gan committed:
docs: added `community` modules descriptions (#17827)

API Reference: Several `community` modules (like
[adapter](https://api.python.langchain.com/en/latest/community_api_reference.html#module-langchain_community.adapters)
module) are missing descriptions. It happens when langchain was split to
the core, langchain and community packages.
- Copied module descriptions from other packages
- Fixed several descriptions to the consistent format.

eon01 committed:
docs: fix a small typo (#17859)

Just a small typo

michaelfeil committed:
community[minor]: infinity embedding local option (#17671)

**drop-in-replacement for sentence-transformers
inference.**

https://github.com/langchain-ai/langchain/discussions/17670

tldr from the discussion above -> around a 4x-22x speedup over using
SentenceTransformers / huggingface embeddings. For more info:
https://github.com/michaelfeil/infinity (pure-python dependency)

---------

Co-authored-by: Erick Friis <erick@langchain.dev>

ehude committed:
community[patch]: Bug Neo4j VectorStore when having multiple indexes the sort is not working and the store that returned is random (#17396)

Bug fix: when having multiple indexes the sort is not working and the
store that returned is random.
The following small fix resolves the issue.

baskaryan committed:
docs: fix api build (#17898)

jacoblee93 committed:
👥 Update LangChain people data (#17900)

👥 Update LangChain people data

---------

Co-authored-by: github-actions <github-actions@github.com>

cbornet committed:
docs: Add docstring to AstraDBStore (#17793)

IANTHEREAL committed:
community[minor]: Add tidb loader support (#17788)

This pull request support loading data from TiDB database with
Langchain.

A simple usage:
```
from  langchain_community.document_loaders import TiDBLoader

CONNECTION_STRING = "mysql+pymysql://root@127.0.0.1:4000/test"

QUERY = "select id, name, description from items;"
loader = TiDBLoader(
    connection_string=CONNECTION_STRING,
    query=QUERY,
    page_content_columns=["name", "description"],
    metadata_columns=["id"],
)
documents = loader.load()
print(documents)
```

vsxd committed:
docs: add helpful comments to sparkllm.py (#17774)

Adding helpful comments to sparkllm.py, help users to use ChatSparkLLM
more effectively

cbornet committed:
community[patch]: Add missing async_astra_db_client param to AstraDBChatMessageHistory (#17742)

nelly-hateva committed:
docs: Update Links to resources in the GraphDB QA Chain documentation (#17720)

- **Description:** Update Links to resources in the GraphDB QA Chain
documentation
    - **Issue:** N/A
    - **Dependencies:** N/A
    - **Twitter handle:** N/A

cbornet committed:
docs: Fix AstraDBVectorStore docstring (#17706)

hinthornw committed:
docs: typo (#17710)

ksraunak77 committed:
community[patch]: Added more functions in NetworkxEntityGraph class (#17624)

- **Description:** 
1. Added add_node(), remove_node(), has_node(), remove_edge(),
has_edge() and get_neighbors() functions in
       NetworkxEntityGraph class.

2. Added the above functions in graph_networkx_qa.ipynb documentation.

hasansustcse13 committed:
community[patch]: Return PK in similarity search Document (#17561)

Issue: #17390

Co-authored-by: hasan <hasan@m2sys.com>

leo-gan committed:
docs: sorting `Examples using ...` section (#17588)

The API Reference docs. If the class has a long list of the examples
that works with this class, then the `Examples using` list is [hard to
comprehend](https://api.python.langchain.com/en/latest/llms/langchain_community.llms.openai.OpenAI.html#langchain-community-llms-openai-openai).
If this list is sorted it would be much easier.
- sorting the `Examples using <ClassName>` list

hinthornw committed:
Redirect the expression language guides (#17914)

gradenr committed:
partner: Add groq partner integration and chat model (#17856)

Description: Add a Groq chat model
issue: TODO
Dependencies: groq
Twitter handle: N/A

Commit:
pinecone[patch], docs: PineconeVectorStore, release 0.0.3 (#17896)

13rac1 committed:
community: Bugfix - correct Ollama API path to avoid HTTP 307 (#17895)

Sets the correct /api/generate path, without ending /, to reduce HTTP
requests.

Reference:

https://github.com/ollama/ollama/blob/efe040f8/docs/api.md#generate-request-streaming

Before:

    DEBUG: Starting new HTTP connection (1): localhost:11434
    DEBUG: http://localhost:11434 "POST /api/generate/ HTTP/1.1" 307 0
    DEBUG: http://localhost:11434 "POST /api/generate HTTP/1.1" 200 None

After:

    DEBUG: Starting new HTTP connection (1): localhost:11434
    DEBUG: http://localhost:11434 "POST /api/generate HTTP/1.1" 200 None

polym committed:
langchain_openai[patch]: fix typos in langchain_openai (#17923)

Just a small typo

Commit:
community: use NeuralDB object to initialize NeuralDBVectorStore (#17272)

**Description:** This PR adds an `__init__` method to the
NeuralDBVectorStore class, which takes in a NeuralDB object to
instantiate the state of NeuralDBVectorStore.
**Issue:** N/A
**Dependencies:** N/A
**Twitter handle:** N/A

Commit:
pinecone[patch]: integration test debug (#17960)

Davidkloving committed:
community[patch]: compatibility with SQLAlchemy 1.4.x (#17954)

**Description:**
Change type hint on `QuerySQLDataBaseTool` to be compatible with
SQLAlchemy v1.4.x.

**Issue:**
Users locked to `SQLAlchemy < 2.x` are unable to import
`QuerySQLDataBaseTool`.

closes https://github.com/langchain-ai/langchain/issues/17819

**Dependencies:**
None

MateuszOssGit committed:
docs: update IBM watsonx.ai docs (#17932)

- **Description:** Update IBM watsonx.ai docs and add IBM as a provider
docs
- **Dependencies:**
[ibm-watsonx-ai](https://pypi.org/project/ibm-watsonx-ai/),
  - **Tag maintainer:** : 

Please make sure your PR is passing linting and testing before
submitting. Run `make format`, `make lint` and `make test` to check this
locally. ✅

Abe410 committed:
community[patch]: Remove subscript for Result type object (#17823)

Resolved 'TypeError: 'type' object is not subscriptable' by removing
subscription of Result type object

Thank you for contributing to LangChain!

- [x] **PR title**: "Langchain: Resolve type error for SQLAlchemy Result
object in QuerySQLDataBaseTool class"

- **Description:** Resolve type error for SQLAlchemy Result object in
QuerySQLDataBaseTool class

- [x] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, hwchase17.

cbornet committed:
langchain[patch]: Support langchain-astradb AstraDBVectorStore in self-query retriever (#17728)

Co-authored-by: Bagatur <baskaryan@gmail.com>

PankajMoolrajani committed:
templates: Update README.md - Fixing a typo (#17689)

- **Description:** PR to fix typo in readme
    - **Issue:** typo in readme
    - **Dependencies:** no
    - **Twitter handle:** p_moolrajani

deanbear committed:
docs: Rerun the Tongyi Qwen model to fix incorrect responses. (#17693)

This PR updates the docs of Tongyi Qwen model. 
1. fix the previously incorrect responses of the Tongyi Qwen.
2. rewrite the case with LCEL.

baskaryan committed:
community[patch]: fix lint (#17984)

Commit:
docs: logo (#17972)

leo-gan committed:
docs: partner package list (#17978)

Updated partner package list

baskaryan committed:
core[minor], openai[minor], langchain[patch]: BaseLanguageModel.with_structured_output #17302)

```python
class Foo(BaseModel):
  bar: str

structured_llm = ChatOpenAI().with_structured_output(Foo)
```

---------

Co-authored-by: Erick Friis <erick@langchain.dev>

baskaryan committed:
langchain[minor]: openai tools structured_output_chain (#17296)

Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>

leo-gan committed:
docs: robocorpai dosctrings (#17968)

Added missing docstrings

---------

Co-authored-by: Bagatur <22008038+baskaryan@users.noreply.github.com>

Commit:
docs, templates: update schema imports to core (#17885)

- chat models, messages
- documents
- agentaction/finish
- baseretriever,document
- stroutputparser
- more messages
- basemessage
- format_document
- baseoutputparser

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

averikitsch committed:
docs: Update Google Provider documentation (#17970)

**Description:** Clean up Google product names and fix document loader
section
**Issue:** NA
**Dependencies:** None

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

mattgotteiner committed:
docs: Update Azure Search vector store notebook (#17901)

- **Description:** Update the Azure Search vector store notebook for the
latest version of the SDK

---------

Co-authored-by: Matt Gotteiner <[email protected]>

chadj2 committed:
community[minor]: add Kinetica LLM wrapper (#17879)

**Description:** Initial pull request for Kinetica LLM wrapper
**Issue:** N/A
**Dependencies:** No new dependencies for unit tests. Integration tests
require gpudb, typeguard, and faker
**Twitter handle:** @chad_juliano

Note: There is another pull request for Kinetica vectorstore. Ultimately
we would like to make a partner package but we are starting with a
community contribution.

Commit:
docs: Added `langchainhub` install and fixed typo (#17985)

Co-authored-by: Erick Friis <erick@langchain.dev>

mackong committed:
community[patch]: callback before yield for _stream/_astream (#17907)

- Description: callback on_llm_new_token before yield chunk for
_stream/_astream for some chat models, make all chat models in a
consistent behaviour.
- Issue: N/A
- Dependencies: N/A

Commit:
infra: ci end check (#17986)

jorgevee committed:
docs: langchain-anthropic README updates (#17684)

# PR Message

- **Description:** This PR adds a README file for the Anthropic API in
the `libs/partners` folder of this repository. The README includes:
  - A brief description of the Anthropic package
  - Installation & API instructions
  - Usage examples
  
- **Issue:**
[17545](https://github.com/langchain-ai/langchain/issues/17545)
  
- **Dependencies:** None

Additional notes:
This change only affects the docs package and does not introduce any new
dependencies.

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

Commit:
infra: ci end check, consolidation (#17987)

Consolidates CI checks into check_diffs.yml in order to properly
consolidate them into a single success status

Issac-Lopez committed:
Update quickstart.mdx (#17659)

https://github.com/langchain-ai/langchain/issues/17657

Thank you for contributing to LangChain!

Checklist:

- [ ] PR title: Please title your PR "package: description", where
"package" is whichever of langchain, community, core, experimental, etc.
is being modified. Use "docs: ..." for purely docs changes, "templates:
..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"
- [ ] PR message: **Delete this entire template message** and replace it
with the following bulleted list
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!
- [ ] Pass lint and test: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified to check that you're
passing lint and testing. See contribution guidelines for more
information on how to write/run tests, lint, etc:
https://python.langchain.com/docs/contributing/
- [ ] Add tests and docs: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, hwchase17.

Commit:
community[patch]: Skip `OpenAIWhisperParser` extremely small audio chunks to avoid api error (#11450)

**Description**
This PR addresses a rare issue in `OpenAIWhisperParser` that causes it
to crash when processing an audio file with a duration very close to the
class's chunk size threshold of 20 minutes.

**Issue**
#11449

**Dependencies**
None

**Tag maintainer**
@agola11 @eyurtsev 

**Twitter handle**
leonardodiegues

---------

Co-authored-by: Leonardo Diegues <leonardo.diegues@grupofolha.com.br>
Co-authored-by: Bagatur <baskaryan@gmail.com>

reidfalconer committed:
langchain[patch]: return formatted SPARQL query on demand (#11263)

- **Description:** Added the `return_sparql_query` feature to the
`GraphSparqlQAChain` class, allowing users to get the formatted SPARQL
query along with the chain's result.
  - **Issue:** NA
  - **Dependencies:** None

Note: I've ensured that the PR passes linting and testing by running
make format, make lint, and make test locally.

I have added a test for the integration (which relies on network access)
and I have added an example to the notebook showing its use.

baskaryan committed:
core[patch]: Release 0.1.26 (#17990)

Commit:
infra: ci failure job (#17989)

Commit:
infra: skip codespell ambr (#17992)

baskaryan committed:
openai[patch]: Release 0.0.7 (#17993)

baskaryan committed:
community[patch]: Release 0.0.22 (#17994)

baskaryan committed:
lanchain[patch]: Release 0.1.9 (#17999)

liugddx committed:
community:  Fix SparkLLM  error (#18015)

Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core,
experimental, etc. is being modified. Use "docs: ..." for purely docs
changes, "templates: ..." for template changes, "infra: ..." for CI
changes.
  - Example: "community: add foobar LLM"

- **Description:** fix SparkLLM  error
- **Issue:** the issue # it fixes, if applicable
- **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!

baskaryan committed:
community[patch]: Release 0.0.23 (#18027)

nicoloboschi committed:
community: fix openai streaming throws 'AIMessageChunk' object has no attribute 'text' (#18006)

After upgrading langchain-community to 0.0.22, it's not possible to use
openai from the community package with streaming=True
```
  File "/home/runner/work/ragstack-ai/ragstack-ai/ragstack-e2e-tests/.tox/langchain/lib/python3.11/site-packages/langchain_community/chat_models/openai.py", line 434, in _generate
    return generate_from_stream(stream_iter)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/ragstack-ai/ragstack-ai/ragstack-e2e-tests/.tox/langchain/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 65, in generate_from_stream
    for chunk in stream:
  File "/home/runner/work/ragstack-ai/ragstack-ai/ragstack-e2e-tests/.tox/langchain/lib/python3.11/site-packages/langchain_community/chat_models/openai.py", line 418, in _stream
    run_manager.on_llm_new_token(chunk.text, chunk=cg_chunk)
                                 ^^^^^^^^^^
AttributeError: 'AIMessageChunk' object has no attribute 'text'
```

Fix regression of https://github.com/langchain-ai/langchain/pull/17907 
**Twitter handle:** @nicoloboschi

Commit:
community[patch]: BaseLLM typing in init (#18029)

baskaryan committed:
community[patch]: Release 0.0.24 (#18038)

Commit:
infra: CI success for partner packages (#18037)

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core,
experimental, etc. is being modified. Use "docs: ..." for purely docs
changes, "templates: ..." for template changes, "infra: ..." for CI
changes.
  - Example: "community: add foobar LLM"


- [ ] **PR message**: ***Delete this entire checklist*** and replace
with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, hwchase17.

Commit:
infra: CI success for partner packages 2 (#18043)

Commit:
docs: recommend lambdas over runnablebranch (#18033)

benjibc committed:
fireworks[patch]: Add Fireworks partner packages (#17694)


---------

Co-authored-by: Erick Friis <erick@langchain.dev>

DannyMac180 committed:
exa[patch]: update readme (#18047)

Commit:
openai[patch]: remove numpy dep (#18034)

Commit:
docs: fireworks fixes (#18056)

Commit:
infra: simplify and fix CI for docs-only changes (#18058)

Current success check will fail on docs-only changes

Commit:
docs: fireworks tool calling docs (#18057)

baskaryan committed:
openai[patch]: refactor with_structured_output (#18052)

- make schema Optional with default val None, since in json_mode you
don't need it if not parsing to pydantic
- change return_type -> include_raw
- expand docstring examples

eyurtsev committed:
core[patch]: Runnable with message history to use add_messages (#17958)

This PR updates RunnableWithMessageHistory to use add_messages which
will save on round-trips for any chat
history abstractions that implement the optimization. If the
optimization isn't
implemented, add_messages automatically invokes add_message serially.

kylehh committed:
community[minor]: Add async_client for Anyscale Chat model (#18050)

Add `async_client` for Anyscale Chat_model

leo-gan committed:
experimental: docstrings update (#18048)

Added missed docstrings. Formatted docsctrings to the consistent format.

2jimoo committed:
community: Add document manager and mongo document manager (#17320)

- **Description:** 
    - Add DocumentManager class, which is a nosql record manager. 
- In order to use index and aindex in
libs/langchain/langchain/indexes/_api.py, DocumentManager inherits
RecordManager.
    - Also I added the MongoDB implementation of Document Manager too.
  - **Dependencies:** pymongo, motor
  
<!-- Thank you for contributing to LangChain!

Please title your PR "<package>: <description>", where <package> is
whichever of langchain, community, core, experimental, etc. is being
modified.

Replace this entire comment with:
- **Description:** Add DocumentManager class, which is a no sql record
manager. To use index method and aindex method in indexes._api.py,
Document Manager inherits RecordManager.Add the MongoDB implementation
of Document Manager.
  - **Dependencies:** pymongo, motor

Please make sure your PR is passing linting and testing before
submitting. Run `make format`, `make lint` and `make test` from the root
of the package you've modified to check this locally.

See contribution guidelines for more information on how to write/run
tests, lint, etc: https://python.langchain.com/docs/contributing/

If you're adding a new integration, please include:
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.

If no one reviews your PR within a few days, please @-mention one of
@baskaryan, @eyurtsev, @hwchase17.
 -->

---------

Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>

jacoblee93 committed:
docs[patch]: Remove redundant Pinecone import (#18079)

CC @efriis

BeatrixCohere committed:
docs: Add Cohere examples in documentation  (#17794)

- Description: Add cohere examples to documentation 
- Issue:N/A
- Dependencies: N/A

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

maximeperrindev committed:
community[patch]: fix llama index imports and fields access (#17870)

- **Description:** Fixing outdated imports after v0.10 llama index
update and updating metadata and source text access
  - **Issue:** #17860
  - **Twitter handle:** @maximeperrin_

---------

Co-authored-by: Maxime Perrin <mperrin@doing.fr>

cbornet committed:
astradb: Add AstraDBChatMessageHistory to langchain-astradb package (#17732)

Co-authored-by: Bagatur <baskaryan@gmail.com>

bhalder committed:
community[minor] : adds callback handler for Fiddler AI (#17708)

**Description:**  Callback handler to integrate fiddler with langchain. 
This PR adds the following -

1. `FiddlerCallbackHandler` implementation into langchain/community
2. Example notebook `fiddler.ipynb` for usage documentation

[Internal Tracker : FDL-14305]

**Issue:** 
NA

**Dependencies:** 
- Installation of langchain-community is unaffected.
- Usage of FiddlerCallbackHandler requires installation of latest
fiddler-client (2.5+)

**Twitter handle:** @fiddlerlabs @behalder

Co-authored-by: Barun Halder <barun@fiddler.ai>

kylehh committed:
community[patch]: Remove model limitation on Anyscale LLM (#17662)

**Description:** Llama Guard is deprecated from Anyscale public
endpoint.
**Issue:** Change the default model. and remove the limitation of only
use Llama Guard with Anyscale LLMs
Anyscale LLM can also works with all other Chat model hosted on
Anyscale.
Also added `async_client` for Anyscale LLM

mattgotteiner committed:
docs: update azure search langchain notebook (#18053)

**Description:** Update the azure search notebook to have more
descriptive comments, and an option to choose between OpenAI and
AzureOpenAI Embeddings

---------

Co-authored-by: Matt Gotteiner <[email protected]>
Co-authored-by: Bagatur <baskaryan@gmail.com>

dokato committed:
langchain[patch]: Make BooleanOutputParser more robust to non-binary responses (#17810)

- **Description:** I encountered this error when I tried to use
LLMChainFilter. Even if the message slightly differs, like `Not relevant
(NO)` this results in an error. It has been reported already here:
https://github.com/langchain-ai/langchain/issues/. This change hopefully
makes it more robust.
- **Issue:**  #11408 
- **Dependencies:** No
- **Twitter handle:** dokatox

DannyMac180 committed:
docs: Additional examples for partners/exa README (#18081)

**Description:** Add additional examples for other modules to
partners/exa README
**Issue:** #17545
**Dependencies:** None
**Twitter handle:** @DannyMcAteer8

---------

Co-authored-by: Daniel McAteer <danielmcateer@Daniels-MBP.attlocal.net>
Co-authored-by: Daniel McAteer <danielmcateer@Daniels-MacBook-Pro.local>

cbornet committed:
astradb[patch]: Use astrapy's upsert_one method in AstraDBStore (#18063)

As `upsert` is deprecated

nelly-hateva committed:
community[patch]: Fix GraphSparqlQAChain so that it works with Ontotext GraphDB (#15009)

- **Description:** Introduce a new parameter `graph_kwargs` to
`RdfGraph` - parameters used to initialize the `rdflib.Graph` if
`query_endpoint` is set. Also, do not set
`rdflib.graph.DATASET_DEFAULT_GRAPH_ID` as default value for the
`rdflib.Graph` `identifier` if `query_endpoint` is set.
  - **Issue:** N/A
  - **Dependencies:** N/A
  - **Twitter handle:** N/A

cbornet committed:
community[patch]: Fix GenericRequestsWrapper _aget_resp_content must be async (#18065)

There are existing tests in
`libs/community/tests/unit_tests/tools/requests/test_tool.py`

Commit:
anthropic[minor]: package move (#17974)

Commit:
google-genai, google-vertexai: move to langchain-google (#17899)

These packages have moved to
https://github.com/langchain-ai/langchain-google

Left tombstone readmes incase anyone ends up at the "Source Code" link
from old pypi releases. Can keep these around for a few months.

Commit:
docs: api docs for external repos (#17904)

Stacked on google removal PR. Will make google continue to show up in
API docs even from external repo

rongchenlin committed:
docs: Fix the bug in MongoDBChatMessageHistory notebook (#18128)

I tried to configure MongoDBChatMessageHistory using the code from the
original documentation to store messages based on the passed session_id
in MongoDB. However, this configuration did not take effect, and the
session id in the database remained as 'test_session'. To resolve this
issue, I found that when configuring MongoDBChatMessageHistory, it is
necessary to set session_id=session_id instead of
session_id=test_session.

Issue: DOC: Ineffective Configuration of MongoDBChatMessageHistory for
Custom session_id Storage

previous code：
```python
chain_with_history = RunnableWithMessageHistory(
    chain,
    lambda session_id: MongoDBChatMessageHistory(
        session_id="test_session",
        connection_string="mongodb://root:Y181491117cLj@123.56.224.232:27017",
        database_name="my_db",
        collection_name="chat_histories",
    ),
    input_messages_key="question",
    history_messages_key="history",
)
config = {"configurable": {"session_id": "mmm"}}
chain_with_history.invoke({"question": "Hi! I'm bob"}, config)
```

![image](https://github.com/langchain-ai/langchain/assets/83388493/c372f785-1ec1-43f5-8d01-b7cc07b806b7)


Modified code:
```python
chain_with_history = RunnableWithMessageHistory(
    chain,
    lambda session_id: MongoDBChatMessageHistory(
        session_id=session_id,   # here is my modify code
        connection_string="mongodb://root:Y181491117cLj@123.56.224.232:27017",
        database_name="my_db",
        collection_name="chat_histories",
    ),
    input_messages_key="question",
    history_messages_key="history",
)
config = {"configurable": {"session_id": "mmm"}}
chain_with_history.invoke({"question": "Hi! I'm bob"}, config)
```

Effect after modification (it works)：


![image](https://github.com/langchain-ai/langchain/assets/83388493/5776268c-9098-4da3-bf41-52825be5fafb)

simonschmidt committed:
langchain: Import from langchain_core in langchain.smith to avoid deprecation warning (#18129)

Avoids deprecation warning that triggered at import time, e.g. with
`python -c 'import langchain.smith'`


/opt/venv/lib/python3.12/site-packages/langchain/callbacks/__init__.py:37:
LangChainDeprecationWarning: Importing this callback from langchain is
deprecated. Importing it from langchain will no longer be supported as
of langchain==0.2.0. Please import from langchain-community instead:

    `from langchain_community.callbacks import base`.

To install langchain-community run `pip install -U langchain-community`.

keenborder786 committed:
langchain[patch]: Update doc-string for a method in ConversationBufferWindowMemory (#18090)

A minor doc fix stated in #18080

hwchase17 committed:
add run name for query constructor (#18101)

Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>

nfcampos committed:
Add BaseMessage.id (#17835)

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core,
experimental, etc. is being modified. Use "docs: ..." for purely docs
changes, "templates: ..." for template changes, "infra: ..." for CI
changes.
  - Example: "community: add foobar LLM"


- [ ] **PR message**: ***Delete this entire checklist*** and replace
with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, hwchase17.

Commit:
docs: anthropic partner package docs (#18109)

nfcampos committed:
Fix bug with using configurable_fields after configurable_alternatives (#18139)

Closes #17915

nfcampos committed:
Improve runnable generator error messages (#18142)

h/t @hinthornw 

Thank you for contributing to LangChain!

- [ ] **PR title**: "package: description"
- Where "package" is whichever of langchain, community, core,
experimental, etc. is being modified. Use "docs: ..." for purely docs
changes, "templates: ..." for template changes, "infra: ..." for CI
changes.
  - Example: "community: add foobar LLM"


- [ ] **PR message**: ***Delete this entire checklist*** and replace
with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
- **Twitter handle:** if your PR gets announced, and you'd like a
mention, we'll gladly shout you out!


- [ ] **Add tests and docs**: If you're adding a new integration, please
include
1. a test for the integration, preferably unit tests that do not rely on
network access,
2. an example notebook showing its use. It lives in
`docs/docs/integrations` directory.


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test`
from the root of the package(s) you've modified. See contribution
guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional
ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in
langchain.

If no one reviews your PR within a few days, please @-mention one of
baskaryan, efriis, eyurtsev, hwchase17.

baskaryan committed:
infra: create api rst for specific pkg (#18144)

Example: create rst for libs/core only
```bash
poetry run python docs/api_reference/create_api_rst.py core
```

baskaryan committed:
core[patch], langchain[patch], templates: move openai functions parsers to core (#18060)

![Screenshot 2024-02-23 at 7 48 03
PM](https://github.com/langchain-ai/langchain/assets/22008038/e5540c4d-0020-4ece-869f-ae19db2a1f3f)

bhalder committed:
docs [patch] : fix import to use community path for handler in fiddler notebook (#18140)

**Description:** Update the example fiddler notebook to use community
path, instead of langchain.callback
**Dependencies:** None
**Twitter handle:** @bhalder

Co-authored-by: Barun Halder <barun@fiddler.ai>

lgabs committed:
[docs] Update doc-string for buffer_as_messages method in ConversationBufferWindowMemory (#18136)

minor fix stated in #18080

HeidiSteen committed:
Docs:  azuresearch.ipynb (in docs/docs/integrations/vectorstores) -- fixed headings and comments (#18135)

This PR updates azuresearch.ipynb with an edit to the introduction
sentence, consistent heading levels, and disambiguation in code
comments.

Commit:
infra: api docs build commit dir (#18147)

Commit:
infra: api docs setup action location (#18148)

danstam17 committed:
community: Add Laser Embedding Integration (#18111)

- **Description:** Added Integration with Meta AI's LASER
Language-Agnostic SEntence Representations embedding library, which
supports multilingual embedding for any of the languages listed here:
https://github.com/facebookresearch/flores/blob/main/flores200/README.md#languages-in-flores-200,
including several low resource languages
- **Dependencies:** laser_encoders

bgdsh committed:
community: make `SET allow_experimental_[engine]_index` configurabe in vectorstores.clickhouse (#18107)

## Description & Issue
While following the official doc to use clickhouse as a vectorstore, I
found only the default `annoy` index is properly supported. But I want
to try another engine `usearch` for `annoy` is not properly supported on
ARM platforms.
Here is the settings I prefer:

``` python
settings = ClickhouseSettings(
    table="wiki_Ethereum",
    index_type="usearch",  # annoy by default
    index_param=[],
)
```
The above settings do not work for the command `set
allow_experimental_annoy_index=1` is hard-coded.
This PR will make sure the experimental feature follow the `index_type`
which is also consistent with Clickhouse's naming conventions.

baskaryan committed:
langchain[patch], core[patch], openai[patch], fireworks[minor]: ChatFireworks.with_structured_output (#18078)

<img width="1192" alt="Screenshot 2024-02-24 at 3 39 39 PM"
src="https://github.com/langchain-ai/langchain/assets/22008038/1cf74774-a23f-4b06-9b9b-85dfa2f75b63">

am-kinetica committed:
Langchain vectorstore integration with Kinetica (#18102)

- **Description:** New vectorstore integration with the Kinetica
database
  - **Issue:** 
- **Dependencies:** the Kinetica Python API `pip install
gpudb==7.2.0.1`,
  - **Tag maintainer:** @baskaryan, @hwchase17 
  - **Twitter handle:**

---------

Co-authored-by: Chad Juliano <cjuliano@kinetica.com>

jaskirat8 committed:
community: vectorstores.kdbai - Added support for when no docs are present (#18103)

- **Description:** By default it expects a list but that's not the case
in corner scenarios when there is no document ingested(use case:
Bootstrap application).
\
Hence added as check, if the instance is panda Dataframe instead of list
then it will procced with return immediately.

- **Issue:** NA
- **Dependencies:** NA
- **Twitter handle:**  jaskiratsingh1

---------

Co-authored-by: Harrison Chase <hw.chase.17@gmail.com>

matthaigh27 committed:
Experimental: Add other threshold types to SemanticChunker (#16807)

**Description**
Adding different threshold types to the semantic chunker. I’ve had much
better and predictable performance when using standard deviations
instead of percentiles.


![image](https://github.com/langchain-ai/langchain/assets/44395485/066e84a8-460e-4da5-9fa1-4ff79a1941c5)

For all the documents I’ve tried, the distribution of distances look
similar to the above: positively skewed normal distribution. All skews
I’ve seen are less than 1 so that explains why standard deviations
perform well, but I’ve included IQR if anyone wants something more
robust.

Also, using the percentile method backwards, you can declare the number
of clusters and use semantic chunking to get an ‘optimal’ splitting.

---------

Co-authored-by: Harrison Chase <hw.chase.17@gmail.com>

maxjakob committed:
partners: add Elasticsearch package (#17467)

### Description
This PR moves the Elasticsearch classes to a partners package.

Note that we will not move (and later remove) `ElasticKnnSearch`. It
were previously deprecated.
`ElasticVectorSearch` is going to stay in the community package since it
is used quite a lot still.

Also note that I left the `ElasticsearchTranslator` for self query
untouched because it resides in main `langchain` package.

### Dependencies
There will be another PR that updates the notebooks (potentially pulling
them into the partners package) and templates and removes the classes
from the community package, see
https://github.com/langchain-ai/langchain/pull/17468

#### Open question
How to make the transition smooth for users? Do we move the import
aliases and require people to install `langchain-elasticsearch`? Or do
we remove the import aliases from the `langchain` package all together?
What has worked well for other partner packages?

---------

Co-authored-by: Erick Friis <erick@langchain.dev>

hwchase17 committed:
add optimization notebook (#18155)

baskaryan committed:
core[patch]: support JS message serial namespaces (#18151)

baskaryan committed:
mistral[minor]: Function calling and with_structured_output (#18150)

![Screenshot 2024-02-26 at 2 07 06
PM](https://github.com/langchain-ai/langchain/assets/22008038/20cacb47-3b24-45b5-871b-dd169f1acd37)

hwchase17 committed:
core[minor]: move document compressor base (#17910)

baskaryan committed:
core[patch]: Release 0.1.27 (#18159)

baskaryan committed:
openai[patch], mistral[patch], fireworks[patch]: releases 0.0.8, 0.0.5, 0.0.2 (#18186)

hwchase17 committed:
Harrison/add structured output (#18165)

isahers1 committed:
docs: deprecation of OpenAI functions agent, astream_events docstring (#18164)

Co-authored-by: Hershenson, Isaac (Extern) <isaac.hershenson.extern@bayer04.de>
Co-authored-by: Bagatur <baskaryan@gmail.com>

Commit:
update content formatter name and include realtime enum

Commit:
update content formatter name and include realtime enum

nfcampos committed:
openai[patch]: Assign message id in ChatOpenAI (#17837)

thehapyone committed:
community[feat]: Adds LLMLingua as a document compressor (#17711)

**Description**: This PR adds support for using the [LLMLingua project
](https://github.com/microsoft/LLMLingua) especially the LongLLMLingua
(Enhancing Large Language Model Inference via Prompt Compression) as a
document compressor / transformer.

The LLMLingua project is an interesting project that can greatly improve
RAG system by compressing prompts and contexts while keeping their
semantic relevance.

**Issue**: https://github.com/microsoft/LLMLingua/issues/31
**Dependencies**: [llmlingua](https://pypi.org/project/llmlingua/)

@baskaryan

---------

Co-authored-by: Ayodeji Ayibiowu <ayodeji.ayibiowu@getinge.com>
Co-authored-by: Harrison Chase <hw.chase.17@gmail.com>

Commit:
airbyte[patch]: init pkg (#18236)

Commit:
airbyte[patch]: core version 0.1.5 (#18244)

averikitsch committed:
docs: add documentation for Google Cloud database integrations (#18225)

**Description:** add documentation for Google Cloud database
integrations
**Issue:** NA
**Dependencies:** NA

Commit:
merge changes

sachinparyani commented:
closing this one in favor of https://github.com/langchain-ai/langchain/pull/18246
